### PR TITLE
Simplify the ExecutionTaskTracker logic and fix some bugs.

### PR DESCRIPTION
**Pull request by becketqin**

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTask.java [hunk 1]
```diff
@@ -55,10 +56,11 @@
     VALID_TRANSFER.put(ABORTED, Collections.emptySet());
   }
 
-  public ExecutionTask(long executionId, ExecutionProposal proposal) {
-    this.executionId = executionId;
-    this.proposal = proposal;
-    this._state = State.PENDING;
+  public ExecutionTask(long executionId, ExecutionProposal proposal, TaskType type) {
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTask.java]: Should we add a JavaDoc? Maybe we can move single line comments of private variable names to params in here.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java [hunk 2]
```diff
@@ -48,10 +47,10 @@ static String getProposalSummary(GoalOptimizer.OptimizerResult result) {
     int numLeaderMovements = 0;
     long dataToMove = 0;
     for (ExecutionProposal p : result.goalProposals()) {
-      if (p.actionType() == ActionType.REPLICA_MOVEMENT) {
+      if (!p.replicasToAdd().isEmpty() || !p.replicasToRemove().isEmpty()) {
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java]: Do we consider a newly added replica's bytes as `dataToMove`? Should we clarify this with a comment line?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java [hunk 3]
```diff
@@ -292,6 +292,25 @@ private void markTaskState(ExecutionTask task, ExecutionTask.State targetState)
...
+          break;
+        case COMPLETED:
+          task.completed();
+          break;
+        default:
+          throw new IllegalStateException("Cannot mark a task in " + task.state() + " to " + targetState + " state");
+      }
     } else {
       throw new IllegalStateException("Cannot mark a task in " + task.state() + " to " + targetState + " state. The "
                                           + "valid target state are " + task.validTargetState());
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java]: "state are " --> "states are "

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java (hunk 2 above)
Comment (becketqin) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServletUtils.java]: replicasToAdd() includes all the replicas that do not exist in the old replica list. The a newly added replica is already included in the dataToMove.

**Pull request merged**